dbbuild/lib/interval.py

```diff
@@ -0,0 +1,36 @@
+# -------------------------------------------------------------------------------------------------
+# Copyright (c) 2023, DHS.
+#
+# This file is part of MicroHapDB (http://github.com/bioforensics/MicroHapDB) and is licensed under
+# the BSD license: see LICENSE.txt.
+#
+# This software was prepared for the Department of Homeland Security (DHS) by the Battelle National
+# Biodefense Institute, LLC (BNBI) as part of contract HSHQDC-15-C-00064 to manage and operate the
+# National Biodefense Analysis and Countermeasures Center (NBACC), a Federally Funded Research and
+# Development Center.
+# -------------------------------------------------------------------------------------------------
+
+from collections import defaultdict
+from intervaltree import IntervalTree
+
+
+class IntervalIndex:
+    def __init__(self):
+        self.trees = defaultdict(IntervalTree)
+        self.mergeables = dict()
+
+    def add(self, marker):
+        self.trees[marker.chrom][marker.start-1:marker.end] = marker
+
+    def check(self):
+        for chrom, tree in sorted(self.trees.items()):
+            tree.merge_overlaps(data_reducer=lambda locus, marker: locus + [marker], data_initializer=list())
```

Comment:
I've used the `intervaltree` package several times before, but I've never taken advantage of this `merge_overlaps` function. It's very convenient!